src/servenix/client/sendnix.py

```
@@ -77,23 +89,27 @@ def _load_cache(self, cache_file, default):
         :return: The contents of the cache parsed as JSON, or the default.
         :rtype: ``object``
         """
-        full_path = join(self._cache_location, cache_file)
         if self._cache_location is None:
             return default
         elif isdir(self._cache_location):
+            full_path = join(self._cache_location, cache_file)
             if not os.access(self._cache_location, os.W_OK):
                 # Can't access cache. Disable it.
                 logging.warn("Couldn't access cache location {}."
                              .format(self._cache_location))
                 self._cache_location = None
                 return default
-            elif not isfile(join(self._cache_location, cache_file)):
+            elif not isfile(full_path):
                 # Cache exists but file does not exist.
                 return default
             else:
-                # File exists in the cache. Parse it as JSON and return it.
-                with gzip.open(full_path, "rb") as f:
-                    return json.loads(f.read().decode("utf-8"))
+                try:
+                    # File exists in the cache. Parse it as JSON and return it.
+                    with gzip.open(full_path, "rb") as f:
+                        return json.loads(f.read().decode("utf-8"))
+                except Exception:
+                    # Couldn't parse the cache.
```

Can you comment as to why it's ok that we're just blindly ignoring this error?